tools/preload.py

```diff
@@ -230,12 +233,19 @@ def fetch_webapp(app_url, directory=None):
     [appname]/metadata.json
     [appname]/update.webapp (if package_path is defined)
     [appname]/cache/ (if appcache_path is defined)
+
+    If port is > 0 that means the webapp is a hosted on
+    localhost. So we get file from filesystem or from a
+    whatever hosting site. In the latter case, icons url
+    are understood to be relative to manifest.webapp.
```

Comment:
nits:
- "is a hosted on localhost" -> *is hosted.
- "So we get file from filesystem or..." -> "As we cannot access localhost at builtime, we get files from the filesystem or from an hosting site. " (I think it makes things clearer).

Concerning the relative path, you could say "In the latter case, icons url are understood to be relative to manifest.webapp regarding the domain hosting this manifest. This way, FxOS will be able to interpret them as absolute at runtime, when the localhost server will be running." WDYT?

I still have a "hacky" feeling about this, but detailed explanations usually ease these kind of feelings :-)